packages/schematics/angular/service/index.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+// TODO: replace `options: any` with an actual type generated from the schema.
+// tslint:disable:no-any
+import {
+  Rule,
+  Tree,
+  apply,
+  branchAndMerge,
+  chain,
+  filter,
+  mergeWith,
+  move,
+  noop,
+  template,
+  url,
+} from '@angular-devkit/schematics';
+import * as stringUtils from '../strings';
+
+import 'rxjs/add/operator/merge';
+import * as ts from 'typescript';
+import {addProviderToModule} from '../utility/ast-utils';
+import {InsertChange} from '../utility/change';
+import {buildRelativePath} from '../utility/find-module';
+
+function addProviderToNgModule(options: any): Rule {
+  return (host: Tree) => {
+    if (!options.module) {
+      return host;
+    }
+
+    let modulePath;
+    if (options.module) {
```

Comment:
This check is redundant (see line 33)
Once that's removed `modulePath` can be a const instead of being `let` and set on line 42.